src/credentials.rs

```diff
@@ -31,55 +35,99 @@ pub struct Credentials {
     pub registry_tokens: HashMap<RegistryUri, String>,
 }
 
-impl Credentials {
-    fn location() -> eyre::Result<PathBuf> {
-        let home = home::home_dir().wrap_err("Failed to locate home directory")?;
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum LocateError {
+    /// the home folder could not be determined
+    MissingHome,
+}
+
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum ExistsError {
+    /// could not locate the credentials file
+    Locate(#[from] LocateError),
+    /// could not access the filesystem
+    Io(#[from] std::io::Error),
+}
 
-        let home = env::var("BUFFRS_HOME").map(PathBuf::from).unwrap_or(home);
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
+pub enum ReadError {
+    /// could not locate the credentials file
+    Locate(#[from] LocateError),
+    /// could not read the file
+    Io(#[from] std::io::Error),
+    /// could not deserialize the credentials
+    Deserialize(#[from] DeserializationError),
+}
+
+#[derive(Error, Display, Debug)]
+#[non_exhaustive]
+#[allow(missing_docs)]
```

Comment:
Unfortunately not, because if we add a top-level doc-comment it will override the per-variant display implementation. It's another downside of using displaydoc...